packages/core/src/command/hl7v2-subscriptions/hl7-subscriptions-sftp-ingestion/hl7-subscriptions-sftp-ingestion.ts

```diff
@@ -0,0 +1,152 @@
+
+    return new LaHieSftpClient(
+      {
+        host,
+        port,
+        username,
+        password,
+      },
+      replica,
+      overridenLog
+    );
+  }
+
+  static async getLaHiePassword(): Promise<string> {
+    const region = Config.getAWSRegion();
+    const passwordArn = Config.getLaHieIngestionPassword();
+    return await getSecretValueOrFail(passwordArn, region);
+  }
+
+  static getLocalReplica(): ReplicaConfig {
+    return { type: "local", path: this.LOCAL_PATH };
+  }
+
+  static getLaHieReplica(): ReplicaConfig {
+    const bucketName = Config.getLaHieIngestionBucket();
+    return { type: "s3", bucketName };
+  }
+
+  async safeSync(remotePath: string): Promise<string[]> {
+    this.overridenLog(`Syncing from remotePath: ${remotePath}`);
+    try {
+      await this.connect();
+
+      const exists = await this.exists(`${remotePath}/`);
+      if (!exists) {
+        throw new Error("Remote path does not exist");
+      }
+      this.overridenLog("Syncing from remote path to Replica");
+      const fileNames = await this.sync(`${remotePath}/`);
+      return fileNames;
+    } finally {
+      await this.disconnect();
+    }
+  }
+
+  override async syncFileToReplica(content: Buffer, remotePath: string) {
+    this.overridenLog("Found a file to sync to replica");
+    if (this.replica) {
+      try {
+        const replicaPath = this.replica.getReplicaPath(remotePath);
+        this.overridenLog("Decrypting content");
+        const decryptedContent = await this.decryptGpgBinaryWithPrivateKey(content);
+
+        this.overridenLog("Syncing decrypted content to replica");
+        await this.replica.writeFile(replicaPath, decryptedContent);
+
+        this.overridenLog("Parsing content from PSV to Hl7");
+        const psvToHl7Converter = new PsvToHl7Converter(decryptedContent);
+        const identifiedMessages = await psvToHl7Converter.getCxIdPtIdHl7MessageList();
+
+        this.overridenLog("Sending parsed data to webhook sender sqs");
+        await sendToWebhookSender(identifiedMessages);
+      } catch (error) {
+        this.overridenLog(`Error writing file to replica: ${error}`);
+      }
+    }
+  }
+
+  private async decryptGpgBinaryWithPrivateKey(content: Buffer): Promise<Buffer> {
+    return content;
+  }
+}
+
+export async function sendToWebhookSender(identifiedMessages: IdentifiedHl7Message[]) {
+  for (const { cxId, ptId, hl7Message } of identifiedMessages) {
+    const messageId = getMessageUniqueIdentifier(hl7Message);
+    const msgTimestamp = getMessageDatetime(hl7Message);
+    const { messageCode, triggerEvent } = getHl7MessageTypeOrFail(hl7Message);
+
+    const rawDataFileKey = createFileKeyHl7Message({
+      cxId,
+      patientId: ptId,
+      timestamp: msgTimestamp ? msgTimestamp : buildDayjs(Date.now()).toISOString(),
+      messageId,
+      messageCode,
+      triggerEvent,
+    });
+
+    const Hl7NotificationParams: Hl7NotificationSenderParams = {
```

Comment:
variables should never start with a capital letter unless its a class definition.